### PR TITLE
[Port] Makes diagonal movement euclidean

### PR DESCRIPTION
**Pull request by atakiya**
<details>
<summary>
Original PR: Citadel-Station-13/Citadel-Station-13#14911
</summary>

## About The Pull Request
Right now, diagonal movement takes twice as long as orthogonal movement. In other words, in terms of mob movement, SS13 treats the tile grid as a taxicab geometry. The issue here is that much of the game doesn't. Projectiles are euclidean: it takes just as much time to move 10 meters diagonally as it does to move 10 meters orthogonally. The thing is, 10 meters diagonally is just under 7 meters north/south and 7 meters east/west, which always takes the equivalent of 14 orthogonal tile movements for units.

In other words: on diagonals, projectiles move sqrt(2) times as fast as mobs do. This fixes that.
</details>

---

## About The Pull Request
TL;DR - it makes diagonal movement take equally as long as non-diagonal movement, instead of two times more.
See the linked PR for more information

## Why It's Good For The Game
Personally I think it makes sense why this is good - going an equal distance should take the same time.
Going diagonal right now just feels awful and makes no sense.

## Changelog
:cl: Avunia Takiya; Putnam3145@Citadel-Station-13
balance: Moving diagonally now takes the same time (distance wise) as non-diagonal movement.
code: Added a `SQRT_2` define capturing the square root of 2, currently used for mob movement.
/:cl:

### CODE REVIEW

**Review by MissFox0810 — APPROVED** [review]
SQRT all over
